plugins/micromega/zify.ml

```diff
@@ -66,35 +66,49 @@ let is_convertible env evd t1 t2 = Reductionops.(is_conv env evd t1 t2)
 let get_type_of env evd e =
   Tacred.cbv_beta env evd (Retyping.get_type_of env evd e)
 
+(* arguments are dealt with in a second step *)
+
 let rec find_option pred l =
   match l with
   | [] -> raise Not_found
   | e :: l -> ( match pred e with Some r -> r | None -> find_option pred l )
 
-(** [HConstr] is a map indexed by EConstr.t.
-    It should only be used using closed terms.
- *)
+module ConstrMap = struct
+  open Names.GlobRef
+
+  type 'a t = 'a list Map.t
+
+  let add evd h e m =
+    let r = fst (EConstr.destRef evd h) in
```

Comment:
Sorry to get here after the battle ended, but could you move the `destRef` calls outside of the API, asking instead a `GlobRef.t` as argument? This would make clearer the possibility that this function might raise a `DestKO` in the callers below.

Reply:
done

Reply:
There are two more calls below. (Also, while you're at it I think sealing the module type behind a signature would make the intent clear.)